src/components/EditCoursePage/index.jsx

```diff
@@ -0,0 +1,213 @@
+/* eslint-disable camelcase */
+import React from 'react';
+import PropTypes from 'prop-types';
+import Helmet from 'react-helmet';
+
+import EditCourseForm from './EditCourseForm';
+import StatusAlert from '../StatusAlert';
+import LoadingSpinner from '../LoadingSpinner';
+
+
+class EditCoursePage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      startedFetching: false,
+    };
+    this.handleCourseEdit = this.handleCourseEdit.bind(this);
+    this.setStartedFetching = this.setStartedFetching.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.fetchCourseInfo();
+    this.props.fetchCourseOptions();
+    this.setStartedFetching();
+  }
+
+  setStartedFetching() {
+    this.setState({ startedFetching: true });
+  }
+
+  handleCourseEdit(options) {
+    /*
+      Need to do some preprocessing before sending anything to course-discovery.
+      This includes:
+        1. Only including subjects that have values
+        2. Putting the entitlement into an array and also adding in the sku
+          (required for updating price)
+        3. Renaming the image and video fields to correspond to what course-discovery is expecting
+        4. Setting the uuid so we can create the url to send to course-discovery
+    */
+    const {
+      courseInfo: {
+        data: {
+          uuid,
+          entitlements,
+        },
+      },
+    } = this.props;
+    const courseData = options;
+    courseData.subjects = [
+      courseData.subjectPrimary,
+      courseData.subjectSecondary,
+      courseData.subjectTertiary,
+    ].filter(subject => !!subject);
+    courseData.entitlements = entitlements && [{
+      mode: courseData.mode,
+      price: courseData.price,
+      sku: entitlements[0].sku,
+    }];
+    courseData.image = courseData.imageSrc;
+    courseData.video = courseData.videoSrc;
+    courseData.uuid = uuid;
+    return this.props.editCourse(courseData);
+  }
+
+  render() {
+    if (!this.props.courseInfo || !this.props.courseOptions) {
+      return (
+        <StatusAlert
+          id="error"
+          alertType="danger"
+          title="Course Edit Form failed to load: "
+          message="Course information unavailable."
+        />
+      );
+    }
+
+    const {
+      courseInfo,
+      courseInfo: {
+        data: {
+          title,
+          key,
+          short_description,
+          full_description,
+          outcome,
+          subjects,
+          image,
+          prerequisites_raw,
+          level_type,
+          learner_testimonials,
+          faq,
+          additional_information,
+          syllabus_raw,
+          video,
+          entitlements,
+        },
+      },
+      courseOptions,
+    } = this.props;
+
+    // If we want to keep a lot of the logic in the lower return,
+    // we have to do all this weird '&&' logic to make sure the data is there.
+    const keyParts = key && key.split(/\+|\//);
```

Comment:
We should avoid special logic to try and parse the key if we can.  Do we need this key split here?

If we do we need to be cautious, any special logic to parse the Course Key and Run key concerns me.

Will this handle new and old style keys?

Also, if we have no other option and we need to do this we should move this to a method so that later if we need it again we can move it to some common tools.

Reply:
We do need something because we don't want to display the key and want to display only the course number.

It does handle the case of the key being split by `+` and by `/` so I think it handles both styles. Happy to move it to a util function if we think that makes the most sense.

Reply:
Let's at least move it to a method and then if we need it again move it to a utility.

Reply:
I think we should make the decision to either move it straight to a utility or keep it as is since it is only being used once right now. Moving it into a method is more lines of code if it's only being called once.

Reply:
The question is, what is the likelihood we will use this again elsewhere in the code?

If you think it will not, I am fine with it staying where it is.

I am ok with it not moving, but please add a named constant variable for the regex.  Just helps with code readability over time.

Reply:
I'm just going to add it into utils.js just in case and then it can be used anywhere. Do you still think the constant variable is necessary once we have
```
function getCourseNumber(courseKeyFragment) {
  const keyParts = courseKeyFragment.split(/\+|\//);
  return keyParts[keyParts.length - 1];
}
```